integration/pkg/api/middleware/recovery.go

```diff
@@ -0,0 +1,38 @@
+package middleware
+
+import (
+	"net/http"
+	"runtime/debug"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+)
+
+// SecureRecovery returns a middleware that recovers from panics and returns a generic error response
+// without exposing internal details. The full error and stack trace are logged for debugging.
+func SecureRecovery(lggr logger.Logger) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		defer func() {
+			if err := recover(); err != nil {
+				stack := debug.Stack()
+
+				// Log the full error with stack trace for internal debugging
+				lggr.Errorw("Panic recovered",
```

Comment:
Do we have the `Panic` level (is that a thing)? Or is it `Critical`? I guess error is fine too.

Reply:
I thought we had a critical, but it seems not to be supported by the chainlink-common logger. And panicw/fatalw calls the actual panic, so it has to stay as an error level 🤷